### PR TITLE
Adding missing docker and fedora images

### PR DESCRIPTION
**Pull request by ibaldin**
@kthare10 can you check if any more images are missing? I can add into the PR.

### CODE REVIEW

File: fabrictestbed_extensions/fablib/fablib.py [hunk 1]
```diff
@@ -1006,6 +1006,11 @@ def get_image_names(self) -> List[str]:
             "default_ubuntu_20",
             "default_ubuntu_21",
             "default_ubuntu_22",
+            "default_fedora_36",
+            "default_fedora_37",
+            "docker_rocky_8",
+            "docker_ubuntu_20",
+            "docker_ubuntu_22"
         ]
```
Comment (kthare10) [inline fabrictestbed_extensions/fablib/fablib.py:1014]: Images to be added:
- default_debian_11
- default_rocky_9

Images to be removed:
- default_cirros
- default_freebsd_13_zfs
- default_openbsd_7

**Review by kthare10 — APPROVED** [review]
+1

**Comment by sajith** [comment]
@kthare10 @ibaldin I pushed some changes to please CI. Please merge whenever you are ready.

**Comment by ibaldin** [comment]
I kinda thought you were in charge of merging this repo @sajith :-)

**Comment by ibaldin** [comment]
BTW, given that it keeps flagging issues - should we instead of checking the format, just run black to fix the formatting as one of the CI steps here? @sajith

**Comment by sajith** [comment]
@ibaldin Okay, I will go ahead and merge. :-)

I am not sure CI doing the formatting is the right thing to do.  That will require that let the CI workflow make commits on the branch, and maybe that would be trusting it a little too much?  Maybe it is better for the PR author to be in full charge of the commits?

Right now the workflow is given just enough permissions:

https://github.com/fabric-testbed/fabrictestbed-extensions/blob/121195e7ef9d1d123244765bb1b3dc47c1059710/.github/workflows/checks.yml#L11-L12

**Comment by kthare10** [comment]
@sajith @ibaldin - I also like the CI running black as otherwise it sometimes feels like an overkill when you make a small change and it complains on every single PR.

**Comment by sajith** [comment]
Yes, true, it is pretty annoying. This is my fault, and I am sorry. :-)

On the other hand, if CI commits a formatting change on the PR branch, and in the meanwhile I have made another bunch of changes locally, I will need to pull the changes from the repo and maybe deal with the merge conflicts.  That also can be annoying.  I don't know how annoying, but I guess that can happen sometimes.

How about using some [pre-commit](https://pre-commit.com/) hooks?  It will require a configuration file where we will define some "hooks" (such as formatting) which will automatically run on `git commit`.  It will require `pip install pre-commit` and `pre-commit install` in the local dev setup.  Would that be a reasonable alternative?

**Comment by ibaldin** [comment]
I personally prefer a linear history so I always rebase, which means this (running black automatically) should not affect anything - it will just be part of the code change on the merged branch.

**Comment by kthare10** [comment]
Alternatively, we can remove this check from CI and run black on the code on the PR which pushes a release version to pypi.

**Comment by sajith** [comment]
> Alternatively, we can remove this check from CI and run black on the code on the PR which pushes a release version to pypi.

That would be a reasonable alternative if I knew how to make that work.  Which I do not. :-(

I do not know how to write a CI rule that triggers just on release-specific PRs, because they are just like any other PR.  Without the nagging from CI, we eventually will forget that we ever used a code formatting check.  Maybe we can trigger `black --check` when a "release" label is present on the PR, but we are likely to forget to add the label.

The problem is that black's feedback is not high enough in value to be not annoying, although consistently formatted code is nice.  What if we replace both black and isort checks with a linter, which might catch some actual problems, both semantic and stylistic?  We do not use one currently in CI.  Perhaps we should?

https://github.com/astral-sh/ruff is the hot new Python linter, and it is even able to fix some of the problems it reports.  Perhaps we can try ruff and see if it offers a happy middle path?

**Pull request merged**